### PR TITLE
enhancements/etcd/disaster-recovery-with-ceo: add backup proposal

### PR DESCRIPTION
**Pull request by hexfusion**
Backups might eventually be managed by a static pod controller. This proposal sets up a directory structure to make that easier to implement.

The idea here is that each master is prepared for restoring the cluster in possible autonomous fashion.

cc @deads2k @alaypatel07 @retroflexer

### CODE REVIEW

File: enhancements/etcd/disaster-recovery-with-ceo.md [hunk 1]
```diff
@@ -85,8 +85,38 @@ This procedure enables to recover from a situation where the control plane certi
...
+2. On all masters write `/etc/kubernetes/static-pod-manifests/backup-N/backup.env` file containing 3 environmental
+   variables `CREATED`, `ETCD_REVISION` and `APISERVER_REVISION`.
+2. On all masters copy directory `/etc/kubernetes/static-pod-manifests/etcd-pod-${ETCD_REVISION}` to
+  `/etc/kubernetes/static-pod-manifests/backup-N/etcd-pod`.
+3. On all masters take an etcd snapshot `etcdctl snapshot save
+  /etc/kubernetes/static-pod-manifests/backup-N/etcd-data/backup.db`.
+4. On all masters copy directory `/etc/kubernetes/static-pod-manifests/kube-apiserver-pod-${APISERVER_REVISION}`
+  to `/etc/kubernetes/static-pod-manifests/backup-N/kube-apiserver-pod`.
+5. On all masters replace directory `/etc/kubernetes/static-pod-manifests/backups` with a copy of
+  `/etc/kubernetes/static-pod-manifests/backup-N` directory.
```
Comment (retroflexer) [inline enhancements/etcd/disaster-recovery-with-ceo.md]: On #5, would a symoblic link of `/etc/kubernetes/static-pod-manifests/backups` to `/etc/kubernetes/static-pod-manifests/backup-N` be better, perhaps?

Comment (hexfusion) [reply enhancements/etcd/disaster-recovery-with-ceo.md:100]: Probably, the size of the etcd snapshot would not make a copy reasonable. Managing the link would be simpler.

File: enhancements/etcd/disaster-recovery-with-ceo.md [hunk 2]
```diff
@@ -85,8 +85,38 @@ This procedure enables to recover from a situation where the control plane certi
 
 ## Proposal
 
+### Backup
+
+1. On all masters create a new backup revision `/etc/kubernetes/static-pod-manifests/backup-N`.
+2. On all masters write `/etc/kubernetes/static-pod-manifests/backup-N/backup.env` file containing 3 environmental
+   variables `CREATED`, `ETCD_REVISION` and `APISERVER_REVISION`.
```
Comment (hexfusion) [inline enhancements/etcd/disaster-recovery-with-ceo.md:92]: this needs to be extended now as we are managing more static pods resources. I think we should also include OCP version.

Comment (retroflexer) [reply enhancements/etcd/disaster-recovery-with-ceo.md:92]: Agree.

**Pull request closed without merging**